scrapy/spidermiddlewares/offsite.py

```diff
@@ -52,9 +53,18 @@ def get_host_regex(self, spider):
         allowed_domains = getattr(spider, 'allowed_domains', None)
         if not allowed_domains:
             return re.compile('') # allow all by default
+        for domain in allowed_domains:
+            url_pattern = re.compile("^https?://.*$")
```

Comment:
No need to compile the regex on each iteration, I'd compile it as class attribute, or in `__init__` or at least before line 56.

Reply:
@dangra done, regex is now being compiled before the loop